### PR TITLE
Fix/ensure correct rooms in watch rooms

### PR DESCRIPTION
**Pull request by lftakakura**

### CODE REVIEW

File: controller/utils.go [hunk 1]
```diff
@@ -1054,6 +1020,47 @@ func AcquireLock(
...
+
+	lock, err := redisClient.EnterCriticalSection(
+		redisClient.Trace(ctx),
+		lockKey,
+		time.Duration(lockTimeoutMS)*time.Millisecond,
+		0, 0,
+	)
+
+	if err != nil {
+		l.WithError(err).Error("error getting watcher lock")
```
Comment (cscatolini) [inline controller/utils.go]: Not necessarily watcher lock, right?

File: controller/utils.go [hunk 2]
```diff
@@ -1054,6 +1020,47 @@ func AcquireLock(
...
+		return nil, err
+	}
+
+	if lock == nil {
+		l.Warnf("unable to get watcher %s lock %s, maybe some other process has it", schedulerName, lockKey)
```
Comment (cscatolini) [inline controller/utils.go]: same comment here

File: controller/utils.go [hunk 3]
```diff
@@ -1054,6 +1020,47 @@ func AcquireLock(
...
+		return nil, fmt.Errorf("unable to get watcher %s lock %s, maybe some other process has it", schedulerName, lockKey)
```
Comment (cscatolini) [inline controller/utils.go]: ditto

File: controller/utils.go [hunk 4]
```diff
@@ -1070,7 +1077,11 @@ func ReleaseLock(
 		err := redisClient.LeaveCriticalSection(lock)
 		if err != nil {
 			l.WithError(err).Error("error releasing lock. Either wait or remove it manually from redis")
+		} else {
+			l.Debug("lock released")
 		}
+	} else {
+		l.Warn("lock is nil. No lock to release")
```
Comment (cscatolini) [inline controller/utils.go]: should this be warn or debug?

File: watcher/watcher.go [hunk 5]
```diff
@@ -447,12 +443,11 @@ func (w *Watcher) ReportRoomsStatuses() error {
 // WithLock is a helper function that runs a block of code
 // that needs to hold a type of lock to redis
 func (w *Watcher) WithLock(l *logrus.Entry, lockKey string, f func() error) (lockErr, err error) {
-	lock, _, err := controller.AcquireLock(
+	lock, err := controller.AcquireLockOnce(
```
Comment (cscatolini) [inline watcher/watcher.go]: What was the previous behaviour? What is this commit fixing by changing AcquireLock to AcquireLockOnce?

Comment (lftakakura) [reply watcher/watcher.go]: AcquireLock use ticker to retry acquiring lock. AcquireLockOnce does not retry.

As watchRooms already runs periodically with ticker, I think it is not necessary to have another ticker to acquire lock.

Also, workers were getting stuck trying to get lock until lockTimeout

Comment (cscatolini) [reply watcher/watcher.go]: Okay, makes sense!

**Pull request merged**